frontend/simple/views/Join.vue

```diff
@@ -97,36 +97,41 @@
     </section>
 </template>
 <script>
+import {namespace} from '../js/backend/hapi'
 import * as Events from '../../../shared/events'
 import backend from '../js/backend/'
-import * as db from '../js/database'
-import {latestContractState} from '../js/state'
+import { latestContractState } from '../js/state'
 import L from '../js/translations'
+
 export default {
   name: 'Join',
-  mounted () {
-    this.fetchData()
+  async mounted () {
+    let state = await latestContractState(this.$route.query.groupId)
+    if (!state.invitees.find(invitee => invitee === this.$store.state.loggedIn)) {
+      console.log(new Error('Invalid Invitation'))
+      await this.$store.dispatch('deleteMail', this.$route.query.inviteHash)
+      this.$router.push({path: '/mailbox'})
+    }
+    let members = []
+    for (let member of state.members) {
+      let identity = await namespace.lookup(member)
+      let identityState = await latestContractState(identity)
+      members.push({name: member, picture: identityState.attributes.picture})
+    }
+    state.members = members
+    this.contract = state
```

Comment:
This section can be simplified to shave 3 lines by:

- Line 115: No need for empty members array
- Line 116: Use index-based iteration here

```js
for (var i = 0; i < state.members.length; i++) {
  const contractId = await namespace.lookup(state.members[i])
  state.members[i] = await latestContractState(contractId)
}
this.contract = state
```